airflow/providers/pinecone/hooks/pinecone.py

```diff
@@ -135,118 +132,98 @@ def upsert(
         )
 
     @staticmethod
+    def get_pod_spec_obj(environment, replicas, shards, pods, pod_type, metadata_config, source_collection):
+        return PodSpec(
+            environment=environment,
+            replicas=replicas,
+            shards=shards,
+            pods=pods,
+            pod_type=pod_type,
+            metadata_config=metadata_config,
+            source_collection=source_collection,
+        )
+
+    @staticmethod
+    def get_serverless_spec_obj(cloud, region):
+        return ServerlessSpec(cloud=cloud, region=region)
+
     def create_index(
+        self,
         index_name: str,
         dimension: int,
-        index_type: str | None = "approximated",
+        spec: ServerlessSpec | PodSpec,
         metric: str | None = "cosine",
-        replicas: int | None = 1,
-        shards: int | None = 1,
-        pods: int | None = 1,
-        pod_type: str | None = "p1",
-        index_config: dict[str, str] | None = None,
-        metadata_config: dict[str, str] | None = None,
-        source_collection: str | None = "",
```

Comment:
This looks like a breaking change. 
PR is still in WIP but just posting a comment to update the changelog accordingly we decided to release a breaking change.